tests/resources/gateway_plugin_acl_test.go

```diff
@@ -0,0 +1,34 @@
+package tests
+
+import (
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-testing/config"
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+)
+
+func TestGatewayPluginACL(t *testing.T) {
+	t.Parallel()
+	t.Run("update-nullify-fields", func(t *testing.T) {
+		resource.Test(t, resource.TestCase{
+			ProtoV6ProviderFactories: providerFactory,
+			Steps: []resource.TestStep{
+				{
+					Config:          providerConfigUs,
+					ConfigDirectory: config.TestNameDirectory(),
+					Check: resource.ComposeAggregateTestCheckFunc(
+						resource.TestCheckResourceAttr("konnect_gateway_plugin_acl.my_acl", "enabled", "true"),
```

Comment:
```suggestion
						resource.TestCheckResourceAttr("konnect_gateway_plugin_acl.my_acl", "enabled", "true"),
						resource.TestCheckResourceAttr("konnect_gateway_plugin_acl.my_acl", "config.allow", "SOME_VALUE"),
```

We should assert the value exists after the first apply to prove that it is removed in the second apply. Apply this feedback to all tests